### DISCUSSION TITLE
Looking for JAVA Code samples to invoke the Airflow API

### DISCUSSION THREAD

**Discussion by shrgupta-tibco** [apache/airflow#17392] [Q&A]
Hi Team,

We plan to invoke the Airflow API using the JAVA Application. Can you help us with the relevant examples which could help us in getting started with Airflow authentication?

Thanks,
Shruti Gupta

**Comment by potiuk** [comment]
We have in plans (and development) a client that is generated automatically from the open-api specification. There is an open issue https://github.com/apache/airflow/issues/17369 as it does not fully work out-of-the-box but you might try to ask there or try to generate the open-api client yourself similarly as we generate Python API client https://github.com/apache/airflow-client-python (this one is officially released by Airflow community already).

**Reply by shrgupta-tibco** [reply]
Thanks for the reply. Will try and explore more on this.
